unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/NFTShape/NFTShapeFrame.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using DCL.Components;
+using NFTShape_Internal;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public interface INFTShapeFrame
+    {
+        /// <summary>
+        /// Get the shape that represent the frame
+        /// </summary>
+        IShape shape { get; }
+        
+        /// <summary>
+        /// The gameObject that represent the frame
+        /// </summary>
+        GameObject gameObject { get; }
+        
+        /// <summary>
+        /// Set the visibility of the frame
+        /// </summary>
+        /// <param name="isVisible"></param>
+        void SetVisibility(bool isVisible);
+        
+        /// <summary>
+        /// Enable or disable the collider of the frame
+        /// </summary>
+        /// <param name="withCollision"></param>
+        void SetHasCollisions(bool withCollision);
+        
+        /// <summary>
+        /// This set the image and creates the HQ Texture handler
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="url"></param>
+        /// <param name="nftAsset"></param>
+        void SetImage(string name, string url, INFTAsset nftAsset);
+        
+        /// <summary>
+        /// Dispose the frame
+        /// </summary>
+        void Dispose();
+        
+        /// <summary>
+        /// Show that it has been an error loading the frame
+        /// </summary>
+        void FailLoading();
+        
+        /// <summary>
+        /// Update the background of the texture so you can set the same background color as the image 
+        /// </summary>
+        /// <param name="newColor"></param>
+        void UpdateBackgroundColor(UnityEngine.Color newColor);
+    }
+    
+    public class NFTShapeFrame : MonoBehaviour, IShape, INFTShapeLoaderController, INFTShapeFrame
```

Comment:
It needs to be a MonoBehaviour because we have like 20 different frames, and each frame has different shapes, and materials... etc. This way it is easy to assign the material and make differents variants for each NFT.

Also it needs the Update callback to calculate if the image should be HQ or not.

We could went further in the refactor to abstract even more from the mono behaviour but that will require more time and changing the code of the current NFTShape in ECS 6. I think that this is a good tradeoff of time/tech debt.

We should refactor in the feature how the HQ is managed along with that class, but I think that it is better to do it in the future since we are going to have only 1 component instead of 2

Reply:
Yeah, I agree, it's out of the scope of this refactor.

IMO the shape itself shouldn't be a monobehaviour but rather control one using the humble pattern.